File: apps/api/internal/models/media.go
package models

type Media struct {
	ID           int64  `json:"id"`
	Type         string `json:"type"`
	Source       string `json:"source"`
	Path         string `json:"path"`
	ExternalID   string `json:"external_id"`
	Title        string `json:"title"`
	Description  string `json:"description"`
	InstagramURL string `json:"instagram_url"`
	YoutubeURL   string `json:"youtube_url"`
	VimeoURL     string `json:"vimeo_url"`
	PreviewPath  string `json:"preview_path"`
	Width        int    `json:"width"`
	Height       int    `json:"height"`
	Featured     bool   `json:"featured"`
	Position     int64  `json:"position"`
	CreatedAt    string `json:"created_at"`

	// Translations: field → lang → text (e.g. translations.title.ru)
	Translations map[string]map[string]string `json:"translations"`
}

File: apps/site/types/media.ts
export type MediaType = 'photo' | 'video'

export type MediaSource = 'upload' | 'youtube' | 'vimeo'

export interface MediaItem {
  id: number
  type: MediaType
  source: MediaSource
  path: string
  /** tiny blurred placeholder; absent on older records (derived by extension then) */
  thumb?: string
  external_id: string
  title: string
  description: string
  instagram_url: string
  youtube_url: string
  vimeo_url: string
  preview_path: string
  width: number
  height: number
  featured: boolean
  position: number
  created_at: string
  /** field → lang → text */
  translations: Record<string, Record<string, string>>
}

export interface UploadResponse {
  path: string
  kind: string
  thumb: string
  width: number
  height: number
}
